integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/header/HTTPResponseCodeTestCase.java

```diff
@@ -41,69 +41,74 @@
 import org.wso2.esb.integration.common.utils.ESBTestCaseUtils;
 
 public class HTTPResponseCodeTestCase extends ESBIntegrationTest {
+
     private Log log = LogFactory.getLog(HTTPResponseCodeTestCase.class);
-	private HttpServer server = null;
+    private HttpServer server = null;
+    private static int responseCode;
 
     @BeforeClass(alwaysRun = true)
     public void init() throws Exception {
         super.init();
-	    String relativePath = "/artifacts/ESB/synapseconfig/esbjava2283/api.xml";
-	    ESBTestCaseUtils util = new ESBTestCaseUtils();
-	    relativePath = relativePath.replaceAll("[\\\\/]", File.separator);
-	    OMElement apiConfig = util.loadResource(relativePath);
-	    addApi(apiConfig);
+        String relativePath = "/artifacts/ESB/synapseconfig/esbjava2283/api.xml";
+        ESBTestCaseUtils util = new ESBTestCaseUtils();
+        relativePath = relativePath.replaceAll("[\\\\/]", File.separator);
+        OMElement apiConfig = util.loadResource(relativePath);
+        addApi(apiConfig);
     }
 
-	@Test(groups = { "wso2.esb" }, description = "Test different response codes", dataProvider = "getResponseCodes")
-	public void testReturnResponseCode(int responseCode) throws Exception {
-		int port = 8089;
-		server = HttpServer.create(new InetSocketAddress(port), 0);
-		server.createContext("/gettest", new MyHandler());
-		server.setExecutor(null); // creates a default executor
-		server.start();
-		switch (responseCode) {
-			case 200:
-				String contentType = "text/xml";
-				String url = "http://localhost:8480/serviceTest/test";
-				sendRequest(url, 200, contentType);
-			case 404:
-				contentType = "text/html";
-				url = "http://localhost:8480/serviceTest/notfound";
-				sendRequest(url, 404, contentType);
-		}
+    @Test(groups = {"wso2.esb" }, description = "Test whether ESB pass-through responses with different response codes.",
+            dataProvider = "getResponseCodes")
+    public void testReturnResponseCode(
+            int responseCode) throws Exception {
+        HTTPResponseCodeTestCase.responseCode = responseCode;
+        //Starting backend server
+        int port = 8089;
+        server = HttpServer.create(new InetSocketAddress(port), 0);
+        server.createContext("/gettest", new ResponseHandler());
+        server.setExecutor(null); // creates a default executor
+        server.start();
+        //Invoke API deployed in ESB
+        switch (responseCode) {
+        case 404:
+            String contentType = "text/html";
+            String url = "http://localhost:8480/serviceTest/notfound";
```

Comment:
Can't we get url from getApiInvocationURL method rather than hard-coding the url?

Reply:
Done